chef/cookbooks/utils/libraries/service.rb

```diff
@@ -0,0 +1,73 @@
+#
+# Copyright 2017, SUSE Linux GmbH
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#  http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# As we want to override the restart method to either call it or skip it,
+# we just monkey patch the Chef::Provider::Service to override the action_restart
+# which is the common method for all the children Service classes
+class Chef
+  class Provider
+    class Service < Chef::Provider
+      def action_restart
+        # how do we know if its allowed to restart? We use the cookbook name as the key
+
+        # as both keystone and horizon use apache2, they trigger a restart of it when something
```

Comment:
It's not only horizon and keystone - also barbican, ceilometer, aodh and maybe others in the future

Reply:
welp, then that screws my whole plan of making this simple :(

Reply:
Actually in that case we **only** do reloads for the apache service which is ok.

same with horizon, we no longer use any restart for apache, only reloads, and that should be ok i guess?